.circleci/config.yml

```diff
@@ -94,20 +94,41 @@ workflows:
   version: 2
   publish:
     jobs:
-      - checkout
+      - checkout:
+          filters:
+            branches:
+              ignore: master
       - build:
           requires:
             - checkout
-      - publish_versioned:
+      - manual_approval:
+          type: approval
           requires:
             - build
+      - publish_versioned:
+          requires:
+            - manual_approval
           context: CodacyDocker
-      - publish_latest:
+      - tag_version:
           requires:
             - publish_versioned
+
+  publish_master:
```

Comment:
Why do we need this?

Reply:
Two workflows for master an non master. This way we can publish master directly without approval.
The alternative would be to have a lot of filters in the same workflow. Something like this maybe:
```
  publish:
    jobs:
      - checkout:
      - build:
          requires:
            - checkout
      - manual_approval:
          type: approval
          requires:
            - build
          filters:
            branches:
              ignore: master
      - publish_versioned:
          requires:
            - manual_approval
          context: CodacyDocker
          filters:
            branches:
              ignore: master
      - publish_versioned_master:
          requires:
            - build
          context: CodacyDocker
          filters:
            branches:
              only: master
      - tag_version:
          requires:
            - publish_versioned
          filters:
            branches:
              ignore: master
      - tag_version_master:
          requires:
            - publish_versioned_master
          filters:
            branches:
              only: master
      - publish_latest:
          requires:
            - tag_version_master
          context: CodacyDocker
      - tag_version:
          requires:
            - publish_versioned
```

Which do you like the most? Split the workflows seemed cleaner and easier to follow to me.